Add advisory and troubleshooting on non-tls mode for machineid kube (#23575)
gravitational/teleport: docs/pages/machine-id/troubleshooting.mdx

$ tctl bots rm example
$ tctl bots add example --roles=foo,bar,machine-id-db
```

> ## Kubernetes connections are failing with Unable to connect to the server: x509: certificate signed by unknown authority

Reviewer: ```suggestion
## Kubernetes connections are failing with `Unable to connect to the server: x509: certificate signed by unknown authority`
```

$ tctl bots rm example
$ tctl bots add example --roles=foo,bar,machine-id-db
```

## Kubernetes connections are failing with `Unable to connect to the server: x509: certificate signed by unknown authority`

### Symptoms

> A self-hosted cluster is connecting Machine ID to Kubernetes clusters with the following errors.  This can happen for non-TLS configured 

Reviewer 2: Since 'cluster' is overloaded, should we be specific?

```suggestion
A self-hosted Teleport cluster is connecting Machine ID to Kubernetes clusters with the following errors.  This can happen for non-TLS configured 
```

> A self-hosted cluster is connecting Machine ID to Kubernetes clusters with the following errors.  This can happen for non-TLS configured 
> Teleport clusters.

Reviewer 2: Line breaks seems to be random throughout. Should be standardized at ~80 chars. This is one example, but please adjust throughout.

```suggestion
A self-hosted cluster is connecting Machine ID to Kubernetes clusters with the
following errors.  This can happen for non-TLS configured Teleport clusters.
```

Author: Please see the udpates

```bash
E0322 22:53:31.653051    1699 memcache.go:265] couldn't get current server API group list: Get "https://teleport.example.com:443/api?timeout=32s": x509: certificate signed by unknown authority
```

> To confirm the TLS routing mode check the `proxy.tls_routing_enabled` from this command with your proxy address. 
> If the value is `false` then this is a non-TLS routing configuration.
> ```bash
> curl https://teleport.example.com:443/webapi/ping | jq

Reviewer 2: 1. check the `proxy.tls_routing_enabled` _what_? Key? Check the value of that key/config?
2. Move the explanation of the output to after the command, and use a `:` before.
3. Keep newlines around code blocks. If/when (most likely when) we move to tooling with standard MDX rendering, adherence to Markdown standards will matter.

````suggestion
To confirm the TLS routing mode check the value of the `proxy.tls_routing_enabled` key with this command, substituting your proxy address:

```bash
curl https://teleport.example.com:443/webapi/ping | jq
```

If the value is `false` then this is a non-TLS routing configuration.
````

```

### Explanation

> Non-TLS routing configured proxies use specific ports for types of traffic. That requires that a Kubernetes

Reviewer 2: ```suggestion
Proxies configured with non-TLS routing"  use specific ports for various types
of traffic. That requires that a Kubernetes
```

Requires additional line break adjustments for the rest of the paragraph.

> connection use its designated port. Currently Machine ID requires the Kubernetes public address is set

Reviewer 2: ```suggestion
connection use its designated port. Currently Machine ID requires that the Kubernetes public address is set
```

or 

```suggestion
connection use its designated port. Currently Machine ID requires the Kubernetes public address to be set
```

to use the correct port. Otherwise it will use the Proxy web port which can cause these
type of errors.

### Resolution

> The `kube_public_addr` is set within the `proxy_service` by administrators. The proxy

Reviewer 2: Within the `proxy_service` _what_, key? Also, `s/proxy/Teleport Proxy service/`.

Author: Please see update

Reviewer 2: The update does not address these concerns.

Reviewer 2: The update does not address these concerns.

Author: Please see latest

will require a restart after the configuration is updated.

  ```yaml
  proxy_service:
    enabled: true
    kube_listen_addr: 0.0.0.0:3026
    kube_public_addr: teleport.example.com:3026
  ```

> Use the configuration listing from the proxy  to confirm the `kube_public_address` is set 

Reviewer 2: Again, if using "the" name the type of thing you're specifying with `kube_public address`.

Author: Please see update

Reviewer 2: Update does not resolve.

Author: Please see latest

$ tctl bots rm example
$ tctl bots add example --roles=foo,bar,machine-id-db
```

## Kubernetes connections are failing with `Unable to connect to the server: x509: certificate signed by unknown authority`

### Symptoms

A self-hosted Teleport cluster is connecting Machine ID to Kubernetes clusters

> with the following errors.  This can happen for non-TLS configured Teleport clusters.

Reviewer 2: ```suggestion
with the following errors. This can happen for non-TLS configured Teleport clusters.
```

```bash
E0322 22:53:31.653051    1699 memcache.go:265] couldn't get current server API group list: Get "https://teleport.example.com:443/api?timeout=32s": x509: certificate signed by unknown authority
```

To confirm the TLS routing mode check the value of the `proxy.tls_routing_enabled`
key with this command, substituting your proxy address:

```bash
curl https://teleport.example.com:443/webapi/ping | jq
```

If the value is `false` then this is a non-TLS routing configuration.

> ```

Reviewer 2: ```suggestion
```

### Explanation

> Proxies configured with non-TLS routing"  use specific ports for various types

Reviewer 2: ```suggestion
Proxies configured with non-TLS routing use specific ports for various types
```
